ecosystem/sep-0008.md

```diff
@@ -61,107 +72,211 @@ approval_server="https://goat.io/tx_approve"
 approval_criteria="The goat approval server will ensure that transactions are compliant with NFO regulation"
 ```
 
+## Transaction Composition
+
+A transaction that an approval server will approve consists of operations that authorize the holding account, transact the regulated asset, and deauthorize the holding account. If a transactions is built this way, there will be no point where holding accounts have open, authorized trustlines to run unapproved operations because of transaction atomocity.
+
+A sandwich transaction described above can be built by wallets preemptively or by approval servers in order to make it compliant when the [revised](#revised) status is applicable.
+
+Depending on whether issuers want to allow the holding accounts to maintain offers, issuers can leave the accounts in the `AUTHORIZE_TO_MAINTAIN_LIABILITIES_FLAG` state or completely deauthorize the accounts when completing the operation. See [CAP-18] for the fine-grained control of authorization.
+
+Here is an example of the transaction:
+
+```
+Operation 1: AllowTrust op where issuer fully authorizes account A, asset X
+Operation 2: AllowTrust op where issuer fully authorizes account B, asset X
+Operation 3: Payment from A to B
+Operation 4: AllowTrust op where issuer sets account B, asset X to AUTHORIZED_TO_MAINTAIN_LIABILITIES_FLAG state
+Operation 5: AllowTrust op where issuer sets account A, asset X to AUTHORIZED_TO_MAINTAIN_LIABILITIES_FLAG state
+```
```

Comment:
I don't think we should be using the `AUTHORIZED_TO_MAINTAIN_LIABILITIES_FLAG` flag in this example, since a payment op doesn't require either account to maintain liabilities. Maybe we have two examples, one with a payment op where both accounts are completely de-authed, and another that involves a single user who wants to make a sell offer, so the issuer leaves the account with `AUTHORIZED_TO_MAINTAIN_LIABILITIES_FLAG`

Reply:
Done in https://github.com/stellar/stellar-protocol/pull/817/